### PR TITLE
WIP - Updating API

### PR DESCRIPTION
**Pull request by Fraser999**
[Review on Reviewable](https://reviewable.io/reviews/dirvine/routing/105)

### CODE REVIEW

File: src/name_type.rs [hunk 1]
```diff
@@ -59,14 +59,13 @@ macro_rules! convert_to_array {
     }};
 }
 
-/// NameType struct
-///
-/// NameType Struct can be created using the new function by passing, id as its parameter.
+/// NameType can be created using the new function by passing id as its parameter.
+#[derive(Default, Eq, PartialOrd, Ord)]
 pub struct NameType(pub [u8; NAME_TYPE_LEN]);
```
Comment (dirvine) [inline src/name_type.rs]: Reminder : we still need to provide PartialOrd for this type before updating API.

File: src/name_type.rs [hunk 2]
```diff
@@ -59,14 +59,13 @@ macro_rules! convert_to_array {
...
 
 impl NameType {
-   #[allow(dead_code)]
-   fn closer_to_target(lhs: &NameType, rhs: &NameType, target: &NameType) -> bool {
+    #[allow(dead_code)]
```
Comment (dirvine) [inline src/name_type.rs]: can we remove this now please.

File: src/routing_node.rs [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
-// See the Licences for the specific language governing permissions and limitations relating to
-// use of the MaidSafe
-// Software.
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// See the Licences for the specific language governing permissions and limitations relating to use
+// of the MaidSafe Software.
 
 extern crate time;
```
Comment (dirvine) [inline src/routing_node.rs]: this is not needed here `use time` instead. extern crate only needed in main lib.rs

**Comment by dirvine** [comment]
Can we please also add `#[deny(missing_docs)]` as well now.

File: Cargo.toml [hunk 4]
```diff
@@ -19,4 +19,4 @@ sodiumoxide = "*"
 rand = "*"
 message_filter = "*"
```
Comment (rossmuir) [inline Cargo.toml]: Could this be updated to `0.1.1`

File: Cargo.toml [hunk 5]
```diff
@@ -19,4 +19,4 @@ sodiumoxide = "*"
...
 lru_time_cache = "0.1.0"
```
Comment (rossmuir) [inline Cargo.toml]: Should this be updated to `0.1.2` ?

File: src/name_type.rs (hunk 1 above)
Comment (dirvine) [inline src/name_type.rs]: This will help
https://github.com/rust-lang/rust/blob/master/src/libcore/array.rs

**Comment by dirvine** [comment]
Reviewed files:
- index.html @ r1

---

<sup>**[src/name_type.rs, line 45 \[r1\]](https://reviewable.io:443/reviews/dirvine/routing/105#-JnTL4ibfu_fNOStz-T_)** ([raw file](https://github.com/dirvine/routing/blob/fdeb7681c05835cfefe02c3fdbd2a947266d1792/src/name_type.rs#L45)):</sup>
Please see http://stackoverflow.com/questions/29784502/convert-vectors-to-arrays-and-back and remove this macro. 

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/105)

**Comment by dirvine** [comment]
Reviewed files:
- Cargo.toml @ r1

---

<sup>**[src/interface.rs, line 16 \[r1\]](https://reviewable.io:443/reviews/dirvine/routing/105#-JnW667fRzp-MsOTKud9)** ([raw file](https://github.com/dirvine/routing/blob/0b1e76b5425577c1c93b289a3fe444c4b29b9a4a/src/interface.rs#L16)):</sup>
Should these be NameTYpe now ?

---


---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dirvine/routing/105)

**Pull request merged**